tests/Tests/ORM/Internal/UnitOfWork/InsertBatchTest.php

```diff
@@ -0,0 +1,118 @@
+<?php
+
+declare(strict_types=1);
+
+namespace Doctrine\Tests\ORM\Internal\UnitOfWork;
+
+use Doctrine\ORM\EntityManagerInterface;
+use Doctrine\ORM\Id\AssignedGenerator;
+use Doctrine\ORM\Id\IdentityGenerator;
+use Doctrine\ORM\Internal\UnitOfWork\InsertBatch;
+use Doctrine\ORM\Mapping\ClassMetadata;
+use PHPUnit\Framework\Attributes\CoversClass;
+use PHPUnit\Framework\Attributes\Group;
+use PHPUnit\Framework\MockObject\Stub;
+use PHPUnit\Framework\TestCase;
+
+#[CoversClass(InsertBatch::class)]
+#[Group('#11977')]
+final class InsertBatchTest extends TestCase
```

Comment:
missing coverage for handling mixed assigned and generated identifier entities?
so

`new EntityA(), new EntityA(), new EntityC(), EntityA()`
?

Reply:
79cc70a62fce0a9c15cdcbd51fb068f57eeadd16